wikimedia/wikipedia-ios
# Add Push permissions opt in to Notifications Center onboarding

**Phabricator:** https://phabricator.wikimedia.org/T302633 (partially)

### Notes
This is a follow on to https://github.com/wikimedia/wikipedia-ios/pull/4155 that adds the actual device-level system permissions request for Notifications. When the user taps the "Turn on push notifications" button during onboarding, we explicitly ask for system permissions if necessary then silently subscribe to Echo subscriptions without interrupting the user experience.

### Test Steps
This can be tested on Simulator or physical device, but of course Echo subscription/Push will only work on a real device build deployed via TestFlight/the App Store.

1. Uninstall/reinstall app and reset privacy permissions if necessary.
2. When logged in, tap the Notifications Center bell. Confirm you see the onboarding flow.
3. Tap "Turn on push notifications" in Notifications Center panel.
4. Confirm you're asked for system permissions for Notifications. If the user has already authorized this permission, they will not be prompted.
